### DISCUSSION TITLE
Fetching a relationship from an already resolved entity

### DISCUSSION THREAD

**Discussion by nrxus** [prisma/prisma#2351] [Q&A] [answered]
Given I have already retrieved my `parentModel` from the prisma client:

```js
await PrismaClient().parentModel.findOne(/* some query */)
```

And my model has one-to-many relationships to a `childModel`. I now want want to get all of the `childModel`s out of my `parentModel` after the fact (i.e., lazy fetching rather than eager fetch). 

I couldn't find in the docs a good solution for this.

I know that I can do:
```js
await PrismaClient().parentModel.findOne(/* some query */).childModels()
```

but because `childModels()` exist as a method in the `ParentModelClient` rather than the `ParentModel` itself, I haven't been able to find a way to get the `childModels()` after I have resolved the entity. 

I know I could do:
```js
await PrismaClient().childModel.findMany({where: { parentModelId: parentModel.id }}) 
```

But that feels like a lot of boiler plate. Is there a better way?

**Comment by nrxus** [comment]
Bump to see if anyone has any ideas or if maybe this should be a feature request (lazy fetching in the model rather than the client).

**Reply by ryands17** [reply, marked as answer]
Hey @nrxus :wave: 
Currently Prisma 2 only supports eager fetching via the ways that you have mentioned above and also with the `include` option while querying.

Could you open a Feature Request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) stating the above use case?

**Comment by Sytten** [comment]
That would not be too bad to use the fluent API if it was optimized. I would really prefer if Prisma sticks with plain javascript objects IMO. No magic, no problem.

**Comment by ludralph** [comment]
Hi @nrxus,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
